### PR TITLE
Rewrote the dev documentation section

### PR DESCRIPTION
**Pull request by thomasrussellmurphy**
Now the dev instructions walk through installing the framework and dependencies.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,16 +33,26 @@ by experimental testing. Note the unexpected additional `:`.
...
+your development computer. In the root of the repo, run `npm install` to install
+all of the dependencies in `package.json`.
+
+With all the dependencies installed, run `bin/hubot` to start shell interaction.
+You'll see some start up output about where your scripts come from and a prompt.
+Some status or error messages are expected if environment variables are not
+fully configured.
+
+Then you can start interacting with hubot by typing anything that may be picked
+up by a listener. Run `hubot help` (the shell will *not* respond to the name
```
Comment (brenns10) [inline README.md]: Should change `hubot help` to `hubot help .*`.  `hubot help` will send a help link, whereas `hubot help .*` will actually output every command's documentation.

**Comment by thomasrussellmurphy** [comment]
Done. Thanks for catching that.

**Comment by brenns10** [comment]
LGTM!

**Comment by thomasrussellmurphy** [comment]
Ehwot?

Edit: oh. Acronyms.

**Pull request merged**